satpy/modifiers/_crefl_utils.py

```diff
@@ -49,6 +88,443 @@
+        band_luts = [lut_array[idx] for lut_array in self.LUTS]
+        return band_luts
+
+    def _find_coefficient_index(self, wavelength_range, resolution=0):
+        """Return index in to coefficient arrays for this band's wavelength.
+
+        This function search through the `COEFF_INDEX_MAP` dictionary and
+        finds the first key where the nominal wavelength of `wavelength_range`
+        falls between the minimum wavelength and maximum wavelength of the key.
+        `wavelength_range` can also be the standard name of the band. For
+        example, "M05" for VIIRS or "1" for MODIS.
+
+        :param wavelength_range: 3-element tuple of (min wavelength, nominal wavelength, max wavelength)
+        :param resolution: resolution of the band to be corrected
+        :return: index in to coefficient arrays like `aH2O`, `aO3`, etc.
+                 None is returned if no matching wavelength is found
+        """
+        index_map = self.COEFF_INDEX_MAP
+        # Find the best resolution of coefficients
+        for res in sorted(index_map.keys()):
+            if resolution <= res:
+                index_map = index_map[res]
+                break
+        else:
+            raise ValueError("Unrecognized data resolution: {}", resolution)
+        # Find the best wavelength of coefficients
+        if isinstance(wavelength_range, str):
+            # wavelength range is actually a band name
+            return index_map[wavelength_range]
+        for k, v in index_map.items():
+            if isinstance(k, str):
+                # we are analyzing wavelengths and ignoring dataset names
+                continue
+            if k[0] <= wavelength_range[1] <= k[2]:
+                return v
+
+
+class _ABICoefficients(_Coefficients):
+    RG_FUDGE = .55  # This number is what Ralph says "looks good" for ABI/AHI
+    LUTS = [
+        # aH2O
+        np.array([2.4111e-003, 7.8454e-003 * RG_FUDGE, 7.9258e-3, 9.3392e-003, 2.53e-2]),
+        # aO2 (bH2O for other instruments)
+        np.array([1.2360e-003, 3.7296e-003, 177.7161e-006, 10.4899e-003, 1.63e-2]),
+        # aO3
+        np.array([4.2869e-003, 25.6509e-003 * RG_FUDGE, 802.4319e-006, 0.0000e+000, 2e-5]),
+        # taur0
+        np.array([184.7200e-003, 52.3490e-003, 15.8450e-003, 1.3074e-003, 311.2900e-006]),
+    ]
+    # resolution -> wavelength -> coefficient index
+    # resolution -> band name -> coefficient index
+    COEFF_INDEX_MAP = {
+        2000: {
+            (0.450, 0.470, 0.490): 0,  # C01
+            "C01": 0,
+            (0.590, 0.640, 0.690): 2,  # C02
+            "C02": 2,
+            (0.8455, 0.865, 0.8845): 3,  # C03
+            "C03": 3,
+            # (1.3705, 1.378, 1.3855): None,  # C04
+            # "C04": None,
+            (1.580, 1.610, 1.640): 4,  # C05
+            "C05": 4,
+            (2.225, 2.250, 2.275): 5,  # C06
+            "C06": 5
+        },
+    }
```

Comment:
I don't see a value of 1, is this deliberate? in which case maybe the second element of the abi coeffs could be deleted?

Reply:
This looks like it is a bug. There are only 5 values in the LUT right now but comparing to @ralphk11's code on SSEC gitlab it looks like the value for C02 is missing. @ralphk11 any ideas what's going on. For context, the ABI/AHI `taur0` values are:

```
np.array([184.7200e-003, 52.3490e-003, 15.8450e-003, 1.3074e-003, 311.2900e-006]),
```

We have in the code above commented out C04 from being used, but your code shows the following values for `taur0`:

```
taur0 = np.array([184.7200e-003, 132.1000e-003, 52.3490e-003, 15.8450e-003, 1.3074e-003, 311.2900e-006])
```

The 132.1e-3 is missing from our values.

Reply:
Ok so I talked to @ralphk11 on slack. The code I was looking at was for AHI which has the "green" 0.51 band. ABI has C04/1.37um. The coefficients in the crefl code were updated in Satpy to not include the 0.51 band but the index values were never updated. I've now added tests for the other support channels and have fixed the coefficient indexes.